edifice/engine.py

```diff
@@ -397,10 +410,11 @@ def _render(self, component: Component, render_context: _RenderContext):
 
         # Call user provided render function and retrieve old results
         sub_component = component.render()
```

Comment:
We will have to do something different here right? Fundamentally, we are changing the type of `render()` so that instead of returning a `Component` it modifies the `RenderContext` and returns `None`?

Reply:
No, the `render()` method still returns a component or a list of components. It's just that those are obtained by the context manager of `Container`